crates/autopilot/src/run_loop.rs

```diff
@@ -181,155 +181,25 @@ impl RunLoop {
-                    .map(|(index, participant)| {
-                        let is_winner = solutions.len() - index == 1;
-                        let mut settlement = SolverSettlement {
-                            solver: participant.driver.name.clone(),
-                            solver_address: participant.solution.solver().0,
-                            score: Some(Score::Solver(participant.solution.score().get().0)),
-                            ranking: solutions.len() - index,
-                            orders: participant
-                                .solution
-                                .orders()
-                                .iter()
-                                .map(|(id, order)| Order::Colocated {
-                                    id: (*id).into(),
-                                    sell_amount: order.sell.into(),
-                                    buy_amount: order.buy.into(),
-                                })
-                                .collect(),
-                            clearing_prices: participant
-                                .solution
-                                .prices()
-                                .iter()
-                                .map(|(token, price)| (token.0, price.get().into()))
-                                .collect(),
-                            call_data: None,
-                            uninternalized_call_data: None,
-                        };
-                        if is_winner {
-                            settlement.call_data = Some(revealed.calldata.internalized.clone());
-                            settlement.uninternalized_call_data =
-                                Some(revealed.calldata.uninternalized.clone());
-                        }
-                        settlement
-                    })
-                    .collect(),
-            };
-            let competition = Competition {
-                auction_id,
-                winner,
-                winning_score,
-                reference_score,
-                participants,
-                prices: prices
-                    .into_iter()
-                    .map(|(key, value)| (key.into(), value.get().into()))
-                    .collect(),
-                block_deadline,
-                competition_simulation_block,
-                call_data,
-                uninternalized_call_data,
-                competition_table,
-            };
-
-            tracing::trace!(?competition, "saving competition");
-            if let Err(err) = self.persistence.save_competition(&competition).await {
-                tracing::error!(?err, "failed to save competition");
-                return;
-            }
+            let auction_uids = auction.orders.iter().map(|o| o.uid).collect::<HashSet<_>>();
 
             if let Err(err) = self
-                .persistence
-                .save_surplus_capturing_jit_orders_orders(
+                .post_processing(
```

Comment:
nanonit: comment explaining why we don't fire and forget the post processing (to make sure we don't settle orders for which we failed to persist data). Maybe we will revisit this if post-processing turns out to be a bottleneck.